services/QuillLMS/app/models/user.rb

```diff
@@ -662,6 +663,15 @@ def segment_user
     SegmentIntegration::User.new(self)
   end
 
+  # With the introduction of the SALES_CONTACT we now have a sort of
+  # "prospective user" type of user.  These people haven't signed up
+  # through our onboarding flow, but are given a User record so that we
+  # can sync their data to Vitally.  We need to treat these users specially
+  # during auth flows because they haven't actually signed up.
+  def non_authenticating?
```

Comment:
I went back and forth on what to call this, and sort of went with the broader version "just in case", but I think you're right that we might as well keep the scope of this restricted until we actually know we have another example.  There's a good chance we'd want to refactor this if we had additional use cases anyway.